File: sessions/2018/easy/219_contains_duplicate_ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        d = {}
        l = len(nums)
        for i in xrange(l):
            n = nums[i]
            if n not in d:
                d[n] = [i]
            else:
                if i - d[n][-1] <= k:
                    return True
                d[n].append(i)
        return False

File: pia/2020/easy/219_contains_duplicate_ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function (nums, k) {
  let map = {};
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    if (map[num] >= 0 && i - map[num] <= k) {
      return true;
    }
    map[num] = i;
  }
  return false;
};
